Source/Core/Core/HW/DVDInterface.cpp

```diff
@@ -1215,8 +1216,60 @@ DVDCommandResult ExecuteCommand(u32 command_0, u32 command_1, u32 command_2,
 // Simulates the timing aspects of reading data from a disc.
 // Returns the amount of ticks needed to finish executing the command,
 // and sets some state that is used the next time this function runs.
-u64 SimulateDiscReadTime(u64 offset, u32 length)
+//
+// This function does some extra handling for Wii discs on
+// top of what SimulateDiscReadTime does. When using a GC disc,
+// (or rather, when decrypt equals false and no data past
+// 4,7 GB is accessed,) both functions do the same thing.
+u64 SimulateDiscReadTimeWii(u64 offset, u32 length, bool decrypt)
 {
+	u64 end_offset = offset + length;
+
+	if (decrypt)
+	{
+		u64 data_start = VolumeHandler::GetDataStart();
+		offset = offset / 0x7C00 * 0x8000 + data_start;
+		if (length == 0)
+			end_offset = offset;
+		else
+			end_offset = (end_offset + 0x7BFF) / 0x7C00 * 0x8000 + data_start;
+	}
+
+	// Offsets on disc layer 2 are replaced with offsets on disc layer 1
+	// to aid in calculations that need to use physical positions.
+	offset %= WII_DISC_LAYER_SIZE;
+	end_offset %= WII_DISC_LAYER_SIZE;
```

Comment:
Is WII_DISC_LAYER_SIZE the size of an encrypted layer or an unencrypted layer?  Either way, you're using it wrong.